load.go

```diff
@@ -405,6 +405,31 @@ func (s Spec) Validate() error {
 		}
 	}
 
+	var patchErr error
+	for src, patches := range s.Patches {
+		for _, patch := range patches {
+			if isRoot(patch.Path) {
```

Comment:
I think we should drop this if conditional to only check the patch if it `isRoot` and check for all of the patches.
As possible wrong use case could happen something like you specify a http patch source but then you also added the
`Path` in the patches block
```yml
sources:
  src:
    git:
      url: https://github.com/kubernetes/kubernetes.git
      commit: "fake"
      keepGitDir: true
  patch_1:
    http:
      url: https://.../fake.patch
patches:
 src:
 - source: patch_1
    path: fake/path
```
and then:
1. we would check that this source exists 
2.  if the source is a dir. handling `http` and `context` sources.
2.1 if the source is not a dir, then we don't expect the path field 
2.2 if it is a dir, we check that path is not root

Reply:
Just to add on to what Danny is saying, I'm fairly certain the behavior here is not correct as is: validation of a patch source shouldn't be conditional on whether it has a root sub-path, it should always be done

Reply:
Right now, we could patch a source with a non-existent one in the spec and there would be no error if the user also supplied a non-root subpath

Reply:
Yes, I think this was a mistake and I extended this if block around the whole thing.